src/main/java/christmas/domain/EventProcess.java

```diff
@@ -0,0 +1,100 @@
+package christmas.domain;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class EventProcess {
+    private static final int MINIMUM_TOTAL_ORDER_PRICE_FOR_GIFT = 120000;
+    private static final int MINIMUM_TOTAL_ORDER_PRICE_FOR_ALL_EVENT = 10000;
+    private static final int D_DAY_DISCOUNT_AMOUNT_PER_DAY = 100;
+    private static final int D_DAY_DISCOUNT_DEFAULT_AMOUNT = 1000;
+    private static final int FIRST_DAY_NUMBER = 1;
+    private static final int WEEKDAY_DESSERT_DISCOUNT_AMOUNT = 2023;
+    private static final int WEEKEND_MAIN_DISCOUNT_AMOUNT = 2023;
+    private static final int SPECIAL_DISCOUNT_AMOUNT = 1000;
+
+    public EventResult takeAllBenefit(Date date, Order order) {
+        Map<BenefitType, Integer> allBenefit = new HashMap<>();
+        allBenefit.put(BenefitType.GIFT, takeGift(order));
+        allBenefit.put(BenefitType.D_DAY, takeDDayDiscount(date));
+        allBenefit.put(BenefitType.WEEKDAY, takeWeekdayDiscount(date, order));
+        allBenefit.put(BenefitType.WEEKEND, takeWeekendDiscount(date, order));
+        allBenefit.put(BenefitType.SPECIAL, takeSpecialDiscount(date));
+
+        cancelAllBenefitIfMeetCondition(order, allBenefit);
```

Comment:
(개인적인 생각입니다!)

할인을 적용하고 조건에 맞지 않으면 되돌리는 방법보다는 처음부터 조건을 만족하지 않으면 할인을 적용하지 않는 것도 하나의 방법일 것 같아요!

```suggestion
        if (this.isInsufficientTotalOrderPriceForEvent(order)) {
            return EnumSet.allOf(BenefitType.class)
                    .stream()
                    .collect(collectingAndThen(
                            toMap(Function.identity(), (benefit) -> 0),
                            EventResult::new)
                    );
        }
```

위 코드에서 stream을 활용했는데 BenefitType의 모든 원소를 가져온다음 0으로 초기화하고 나온 결과물을 EventResult로 래핑하는 코드입니다 :)

Reply:
미션 동안은 시간이 촉박했어서 stream공부를 하지 못했는데요, 앞으로 공부해서 추천해주신 방법 이해해보도록 할게요 감사합니다!!

Reply:
저도 같은 생각이에요! 다 적용한다음 취소하는 것보다 조건을 먼저 확인해서 할인 적용 유무를 따지는게 더 괜찮은 것 같아요!